Financial app implementation

devalibello opened the pull request:
This pull request includes the full implementation of a financial app. The app grants users to keep track of various bills and payments based on various categories. It includes:

- N+1 solution
- Validations
- Authentication using Devise
- Live deployment on Render
- Integration, Model, and Request tests using RSpec
Kindly find screenshoots of the before and after implementation of eager loading

`WITHOUT EAGER LOADING`

![payments-without-eager-loading](https://github.com/devalibello/budget-app/assets/114155239/3b0851af-0148-43ff-b2dc-87ca22b5fbe1)

`WITH EAGER LOADING`

![payments-with-eager-loading](https://github.com/devalibello/budget-app/assets/114155239/4cebdcfe-166f-4a2b-92ac-b989f1363895)

devalibello committed:
create models

devalibello committed:
Implement all features

devalibello committed:
deploy

devalibello committed:
deploy-3

devalibello committed:
deploy

devalibello committed:
Add gemfile

devalibello committed:
Add gemfile

devalibello committed:
update readme file

divinecharlotte reviewed (changes requested):
Hi @devalibello    , 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## To Highlight 💯 

- Clean code👍 
- Implemented some project requirements   👍 

 




 
![](https://media.giphy.com/media/tKyrc0NTlEcGWdThp4/giphy.gif)
## Required Changes ♻️

- [x] Concerning the video presentation kindly consider making sure your  video contains a clear and objective description about the project in professional English. The topic and purpose of the project are made clear. and also please put in considerartion to highlight about a piece of the code you found  interesting or you are proud of making. to meet the video presantation requirements
- [x] kindly consider having the app's name on the splash screen as requested in the project requirement or as given on the design
- [ ] kindly consider making sure that if a user adds the first  category and go to add transaction on please make sure  the transaction are also displayed as well cause currently they are only being diplsyed only on the second category and remember in the project requirements all the categories transaction should be displayed the same  check on the image below i attached 
- ![now](https://github.com/devalibello/budget-app/assets/60146030/b42d1523-6f39-4232-8d79-61281345caf3)
so please make sure every single transaction added is displayed on the UI in the right place as well

- [x] kindly consider making sure For a given category, the list of transactions is presented ordered by the most recent to meet the project requirements cause currently they are being shown in the inverse order check he mage below for  more clarrification
- 
![no](https://github.com/devalibello/budget-app/assets/60146030/accba70d-8a4e-4318-98d3-a89fd3558d9d)

- [x] in summary before you submit consider going back and the requirement per requirement in order to not miss somethig else and that will reduce the amout of times to be given feedback of what to correct

_Check the comments under the review._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

devalibello committed:
fix order of transactions

devalibello committed:
fix linters error

devalibello committed:
fix splash page

devalibello committed:
fix payment options

devalibello committed:
fix payment options

devalibello committed:
fix payment options

devalibello committed:
fix render deployment

devalibello committed:
fix render deployment

devalibello committed:
fix render deployment

devalibello committed:
update video presentation link

devalibello commented:
Hello CR
All previous review comments have been addressed
- Updated video presentation
- Added a splash screen as attached below
![image](https://github.com/devalibello/budget-app/assets/114155239/85f2ee65-d275-4602-917a-30e3d24a35de)
- Fixed the miscalculations using `Category.where(author_id: current_user.id)` to implement the drop-down list of categories
- Payments and Categories are now displayed based on the project requirement using this line of code `<% @payments.order(created_at: :desc).each do |payment| %>`

ShadyShawkat reviewed (approved):
## Hi @devalibello, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 
- Your readme looks professional ✔️ 

![Image](https://media.giphy.com/media/jJQC2puVZpTMO4vUs0/giphy.gif)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
